src/sentry/web/frontend/base.py

```diff
@@ -303,6 +304,14 @@ def dispatch(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         if self.is_sudo_required(request, *args, **kwargs):
             return self.handle_sudo_required(request, *args, **kwargs)
 
+        if (
+            is_using_customer_domain(request)
```

Comment:
ooof.  I don't suppose there's anyway this logic could be rolled into individual subclasses rather than this inspection?  Not going to block if you think it's the best approach.